rust/datafusion/src/physical_plan/sort.rs

```diff
@@ -56,7 +63,12 @@ impl SortExec {
         expr: Vec<PhysicalSortExpr>,
         input: Arc<dyn ExecutionPlan>,
     ) -> Result<Self> {
-        Ok(Self { expr, input })
+        Ok(Self {
+            expr,
+            input,
+            output_rows: SQLMetric::counter("outputRows"),
+            sort_time_nanos: SQLMetric::time_nanos("sortTime"),
```

Comment:
Given Rust's focus on compile time type checking, what would you think about using typed counters rather than `String` keys? 

So make the code look something like:

```suggestion
            output_rows: SQLMetric<OutputRows>::new(),
            sort_time_nanos: SQLMetric<SortTime>::new(),
```

Reply:
Does this imply adding an enum for the metrics? This might limit extensibility for users that want to add custom metrics.